endpoints/openrtb2/auction.go

```diff
@@ -1282,14 +1285,25 @@ func effectivePubID(pub *openrtb.Publisher) string {
 	return pbsmetrics.PublisherUnknown
 }
 
-func validateAccount(cfg *config.Configuration, pubID string) error {
+func (deps *endpointDeps) validateAccount(ctx context.Context, pubID string) (*config.Account, error) {
 	var err error = nil
-	if cfg.AccountRequired && pubID == pbsmetrics.PublisherUnknown {
-		// If specified in the configuration, discard requests that don't come with an account ID.
-		err = error(&errortypes.AcctRequired{Message: fmt.Sprintf("Prebid-server has been configured to discard requests that don't come with an Account ID. Please reach out to the prebid server host.")})
-	} else if _, found := cfg.BlacklistedAcctMap[pubID]; found {
-		// Blacklist account now that we have resolved the value
-		err = error(&errortypes.BlacklistedAcct{Message: fmt.Sprintf("Prebid-server has blacklisted Account ID: %s, please reach out to the prebid server host.", pubID)})
+	var account *config.Account
+	if pubID == pbsmetrics.PublisherUnknown {
+		account = &config.UnknownAccount
+	} else {
+		account, err = deps.acccounts.FetchAccount(ctx, pubID)
 	}
-	return err
+	if account.Disabled {
+		if account.ID == config.UnknownAccount.ID {
+			err = error(&errortypes.AcctRequired{
+				Message: fmt.Sprintf("Prebid-server has been configured to discard requests that don't come with an Account ID. Please reach out to the prebid server host."),
```

Comment:
The "account" ID is pulled from the openRTB "publisher" ID. Probably best to word the messages appropriately to reduce confusion, as there is no explicit "account" field in openRTB.

Reply:
I agree wrt confusion. With PBS' concept of "account" overlapping that of "publisher", where to reference one or the other is a bit of gray area. I've preserved the previous messaging in this case, but I can change it.